flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/statemachine/StateMachineExample.java

```diff
@@ -92,7 +96,29 @@ public static void main(String[] args) throws Exception {
 
 		// create the environment to create streams and configure execution
 		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-		env.enableCheckpointing(5000);
+
+		final String checkpointDir = params.getRequired("checkpointDir");
```

Comment:
We could do this also via the `flink-conf.yaml` and don't make the parameters of the example too complicated.

Reply:
In any case you should update the `usage string` above and end to end tests that depend on this test as well.